File: src/main/scala/jk_5/nailed/plugins/nmm/NmmPlugin.scala
package jk_5.nailed.plugins.nmm

import java.io.File

import com.google.gson.Gson
import jk_5.eventbus.EventHandler
import jk_5.nailed.api.event.mappack.RegisterMappacksEvent
import jk_5.nailed.api.plugin.Plugin
import org.apache.http.client.methods.HttpGet
import org.apache.http.impl.client.HttpClientBuilder
import org.apache.http.util.EntityUtils
import org.apache.logging.log4j.LogManager

object NmmPlugin {
  var instance: NmmPlugin = _
}

@Plugin(id = "Nailed|NMM", name = "Nailed Mappack Manager", version = "1.0.0")
class NmmPlugin {
  NmmPlugin.instance = this

  val cache = new File("nmm-cache")
  val gson = new Gson
  val logger = LogManager.getLogger
  if(!cache.exists()) cache.mkdir()

  @EventHandler
  def registerMappacks(event: RegisterMappacksEvent){
    val httpClient = HttpClientBuilder.create().build()
    try{
      val request = new HttpGet("http://nmm.jk-5.tk/mappacks.json")
      val response = httpClient.execute(request)
      val list = gson.fromJson(EntityUtils.toString(response.getEntity, "UTF-8"), classOf[MappacksList])
      list.mappacks.foreach { e =>
        event.registerMappack(new NmmMappack(e))
      }
      logger.info("Registered " + list.mappacks.size + " nmm mappacks")
    }catch{
      case e: Exception => logger.warn("Was not able to fetch mappacks from nmm.jk-5.tk", e)
    }
  }
}

File: src/main/java/jk_5/nailed/plugins/nmm/NmmPlugin.java
package jk_5.nailed.plugins.nmm;

import com.google.gson.Gson;
import jk_5.eventbus.EventHandler;
import jk_5.nailed.api.event.mappack.RegisterMappacksEvent;
import jk_5.nailed.api.plugin.Plugin;
import org.apache.http.HttpResponse;
import org.apache.http.client.HttpClient;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.impl.client.HttpClientBuilder;
import org.apache.http.util.EntityUtils;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.io.File;

@Plugin(id = "Nailed|NMM", name = "Nailed Mappack Manager", version = "1.0.0")
public class NmmPlugin {

    private static final Logger logger = LogManager.getLogger();
    public static final Gson gson = new Gson();
    public static NmmPlugin instance;
    private final File cache = new File("nmm-cache");

    public NmmPlugin() {
        instance = this;
        if(!cache.exists()){
            cache.mkdir();
        }
    }

    @EventHandler
    public void registerMappacks(RegisterMappacksEvent event){
        HttpClient httpClient = HttpClientBuilder.create().build();
        try{
            HttpGet request = new HttpGet("http://nmm.jk-5.nl/mappacks.json");
            HttpResponse response = httpClient.execute(request);
            MappacksList list = gson.fromJson(EntityUtils.toString(response.getEntity(), "UTF-8"), MappacksList.class);
            for(String mappack : list.mappacks){
                event.registerMappack(new NmmMappack(mappack));
            }
            logger.info("Registered " + list.mappacks.length + " nmm mappacks");
        }catch(Exception e){
            logger.warn("Was not able to fetch mappacks from nmm.jk-5.nl", e);
        }
    }
}
